chunky/src/java/se/llbit/chunky/renderer/Renderer.java

```diff
@@ -17,85 +16,52 @@
-   * time and average samples per second.
+   * Get the friendly name of this renderer.
    */
-  void setOnRenderCompleted(BiConsumer<Long, Integer> listener);
+  String getDisplayName();
 
   /**
-   * Set a listener for frame completion.
+   * The post render callback. This should be run after rendering a frame.
+   * It will return {@code true} if the render should terminate.
    *
-   * @param listener a listener which is called when a frame completes
-   * with the current scene and the current samples per pixel.
-   */
-  void setOnFrameCompleted(BiConsumer<Scene, Integer> listener);
-
-  void setSnapshotControl(SnapshotControl callback);
-
-  void setRenderTask(TaskTracker.Task task);
-
-  void addRenderListener(RenderStatusListener listener);
-
-  void removeRenderListener(RenderStatusListener listener);
-
-  void withBufferedImage(Consumer<BitmapImage> bitmap);
-
-  interface SampleBufferConsumer {
-    void accept(double[] samples, int width, int height);
-  }
-
-  void addSceneStatusListener(SceneStatusListener listener);
-
-  void removeSceneStatusListener(SceneStatusListener listener);
-
-  RenderStatus getRenderStatus();
-
-  /**
-   * Start up the renderer.
+   * Generally the render loop will look like:
+   * {@code
+   *   while (scene.spp < scene.getTargetSpp()) {
+   *     submitTiles(manager, (state, pixel) -> {});
+   *     manager.pool.awaitEmpty();
+   *     scene.spp += 1; // update spp
+   *     if (postRender.getAsBoolean()) break;
+   *   }
+   * }
    *
-   * <p>This should start all worker threads used by the renderer.
+   * Implementation details, this deals with:
+   *  * Checking if the render mode has changed
+   *  * Updating the task-tracker
+   *  * Repainting the canvas
+   *  * Updating the {@code bufferedScene}
    */
-  void start();
+  void setPostRender(BooleanSupplier callback);
 
   /**
-   * Wait for the renderer to terminate. This should only be done
-   * in headless rendering, otherwise the renderer will not automatically
-   * shut down after the render completes.
-   * @throws InterruptedException
+   * This is called when a render is initiated.
+   *
+   * * It should render a frame, merge that frame with {@code manager.bufferedScene}, and update the spp values.
+   * * It should call the post-render callback (set in {@code setPostRender(callback)}) and terminate if it returns {@code true}.
    */
-  void join() throws InterruptedException;
-
-  void withSampleBufferProtected(SampleBufferConsumer consumer);
+  void render(DefaultRenderManager manager) throws InterruptedException;
 
   /**
-   * Shut down the renderer.
-   *
-   * <p>This should interrupt all worker threads used by the renderer.
+   * This is called when the scene is reset. The default implementation does nothing.
+   * This is for {@code Renderer}s which need to export the scene data in some way.
    */
-  void shutdown();
+  default void sceneReset(DefaultRenderManager manager, ResetReason reason) {}
```

Comment:
Add a method to see if the RenderManager should finalize the frames in the postRender callback.